site/docs/spec.md

```diff
@@ -581,10 +596,11 @@ Table metadata consists of the following fields:
 | _optional_ | _optional_ | **`metadata-log`**| A list (optional) of timestamp and metadata file location pairs that encodes changes to the previous metadata files for the table. Each time a new metadata file is created, a new entry of the previous metadata file location should be added to the list. Tables can be configured to remove oldest metadata log entries and keep a fixed-size log of the most recent entries after a commit. |
 | _optional_ | _required_ | **`sort-orders`**| A list of sort orders, stored as full sort order objects. |
 | _optional_ | _required_ | **`default-sort-order-id`**| Default sort order id of the table. Note that this could be used by writers, but is not used when reading because reads use the specs stored in manifest files. |
+|            | _optional_ | **`refs`** | A list of snapshot references, stored as full snapshot reference objects. |
```

Comment:
I think that this PR needs to address compatibility with `current-snapshot-id`. For the duration of v2, writers must still produce `current-snapshot-id`.

Reply:
I see the `current-snapshot-id` field is described as optional, but when checking the `TableMetadataParser` it looks like we always assume the field exists for both v1 and v2. 

https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableMetadataParser.java#L400

Is that an issue with the spec itself?

If this field is required for v1 and v2, then I think there is no concern? Or are you suggesting adding a new column for format v3 which deprecates `current-snapshot-id`?

Reply:
The current snapshot ID is optional because the table may not have a snapshot ID. We can clarify this in the spec and update the parser to return `null`.